### PR TITLE
Tau DQM code relocation from Validation/RecoTau area

### PR DESCRIPTION
**Pull request by amodakgit**
Following the proposal from RECO group, the data certification related Tau DQM code is now moved from **Validation/RecoTau** to  **DQMOffline/PFTau** package. The idea is to separate the genuine data certification code related to Tau DQM from mc validation which is in the Validation/RecoTau package. Common part of the data & mc code are kept in the DQMOffline/PFTau and will also be used by mc validation.
These changes have been approved by TauPOG and you can see the slides here for more details,
https://indico.cern.ch/event/565857/contributions/2285741/attachments/1328124/1994864/TauDQM_29Aug16.pdf

cc: @roger-wolf

### CODE REVIEW

File: Validation/RecoTau/src/TauTagMCValidation.cc [hunk 1]
```diff
@@ -0,0 +1,597 @@
...
+  // ----------------------- Reference product -----------------------------------------------------------------------
+
+  Handle<genCandidateCollection> ReferenceCollection;
+  bool isGen = iEvent.getByToken( refCollectionInputTagToken_, ReferenceCollection );
+
+  Handle<VertexCollection> pvHandle;
+  iEvent.getByToken( primaryVertexCollectionToken_, pvHandle ); //TO-DO
+
+  if (!isGen) {
+    std::cerr << " Reference collection: " << refCollection_ << " not found while running TauTagMCValidation.cc " << std::endl;
```
Comment (dmitrijus) [inline Validation/RecoTau/src/TauTagMCValidation.cc]: Please replace std:cerr with LogWarning. Rule of thumb: never use std:cout/cerr in CMSSW.

**Review by dmitrijus — CHANGES_REQUESTED** [review]

File: Validation/RecoTau/src/TauTagMCValidation.cc (hunk 1 above)
Comment (amodakgit) [inline Validation/RecoTau/src/TauTagMCValidation.cc]: thanks for the suggestion. it is now corrected.

**Comment by dmitrijus** [comment]
+1

**Comment by dmitrijus** [comment]
please test

**Comment by davidlange6** [comment]
Hi @amodakgit - any plans to update this PR?

**Comment by amodakgit** [comment]
Hi David,

Still facing some issues to validate the fix. But hopefully will converge
by next week.

thanks,
atanu


________________________________

**Comment by davidlange6** [comment]
lets resume this PR review in 90x once it is updated.

**Pull request closed without merging**